python/whylogs/api/logger/rolling.py

```diff
@@ -170,12 +187,15 @@ def _get_time_tuple(self) -> time.struct_time:
                 time_tuple = time.localtime(self._current_batch_timestamp + addend)
         return time_tuple
 
-    def _flush(self, profile: DatasetProfile) -> None:
-        if profile is None:
+    def _flush(self, results: ResultSet) -> None:
+        if results is None:
             return
-        if self.skip_empty and profile.is_empty:
-            logger.debug("skip_empty is set. Skipping empty profiles")
+
+        profiles = results.get_writables()
+        if not profiles:
```

Comment:
is this verifying if it's an empty list or None? i think it'd be best to make it explicit

Reply:
It is intentionally doing both, None and [] both get skipped there.